test/components/Modal.test.js

```diff
@@ -0,0 +1,109 @@
+      wrapper.simulate('click');
+      expect(onButtonClick).toHaveProperty('callCount', 1);
+    });
+  });
+
+  describe('next button', () => {
+    const onButtonClick = sinon.spy();
+    const wrapper = shallow(<Next photo={5} onButtonClick={onButtonClick} />);
+    wrapper.simulate('click');
+
+    test('should invoke a handler on click', () => {
+      expect(onButtonClick).toHaveProperty('callCount', 1);
+    });
+
+    test('should point to the next photo', () => {
+      expect(onButtonClick.calledWith('NEXT')).toBe(true);
+    });
+  });
+
+  describe('prev button', () => {
+    const onButtonClick = sinon.spy();
+    const wrapper = shallow(<Prev photo={5} onButtonClick={onButtonClick} />);
+    wrapper.simulate('click');
+
+    test('should invoke a handler on click', () => {
+      expect(onButtonClick).toHaveProperty('callCount', 1);
+    });
+
+    test('should point to the previous photo', () => {
+      expect(onButtonClick.calledWith('PREV')).toBe(true);
+    });
+  });
+
+  describe('photo nav button', () => {
+    test('should invoke a handler on click', () => {
+      const onButtonClick = sinon.spy();
+      const wrapper = shallow(<PhotoNav onButtonClick={onButtonClick} />);
+      wrapper.simulate('click');
+      expect(onButtonClick).toHaveProperty('callCount', 1);
+    });
+
+    test('should have an img tag', () => {
+      const wrapper = shallow(<PhotoNav />);
+      expect(wrapper.find('img')).toHaveLength(1);
+    });
+
+    test('should have an id set to the `photo` prop', () => {
+      const wrapper = shallow(<PhotoNav photo={5} />);
+      expect(wrapper.props()).toHaveProperty('id', 'nav-5');
+    });
+
+    test('should have an href pointing to itself', () => {
+      const wrapper = shallow(<PhotoNav photo={5} />);
+      expect(wrapper.props()).toHaveProperty('href', '#nav-5');
+    });
+  });
+});
+
+describe('photo reel component', () => {
+  test('should have specific subcomponents', () => {
+    const wrapper = shallow(<PhotoReel photos={[]}/>);
+    expect(wrapper.find('FlexRow')).toHaveLength(1);
+    expect(wrapper.find('Next')).toHaveLength(1);
+    expect(wrapper.find('ScrollBox')).toHaveLength(1);
+    expect(wrapper.find('Prev')).toHaveLength(1);
+  });
+
+  test('should dynamically render PhotoNav components', () => {
+    const target = testPhotoData.length;
+    const wrapper = shallow(<PhotoReel photos={testPhotoData}/>);
+    expect(wrapper.find('PhotoNav')).toHaveLength(target);
+  });
+});
+
+describe('modal component', () => {
+  test('should have specific subcomponents', () => {
+    const wrapper = shallow(<Modal photos={testPhotoData} />);
+    expect(wrapper.find('Fullscreen')).toHaveLength(1);
+    expect(wrapper.find('FlexColumn')).toHaveLength(1);
+    expect(wrapper.find('Close')).toHaveLength(1);
+    expect(wrapper.find('PhotoWrapper')).toHaveLength(1);
+    expect(wrapper.find('Description')).toHaveLength(1);
+  });
+});
```

Comment:
@krise3 These are some impressive tests. Really impressed with implementation of using sinon to test the functionality of you button clicks.